app/Filament/Bre/Resources/FieldResource.php

```diff
@@ -76,16 +81,120 @@ public static function form(Form $form): Form
             ]);
     }
 
+    public static function infolist(Infolist $infolist): Infolist
+    {
+        return $infolist
+            ->schema([
+                TextEntry::make('name'),
+                TextEntry::make('label'),
+                TextEntry::make('help_text'),
+                TextEntry::make('breDataType.name')
+                    ->label('Data Type'),
+                TextEntry::make('breDataValidation.name')
+                    ->formatStateUsing(function ($state, $record) {
+                        if (!$record->breDataValidation) {
+                            return '';
+                        }
+
+                        return new HtmlString(
+                            sprintf(
+                                '<a href="%s" style="text-decoration: none; display: inline-block; margin: 2px;">
+                                <span class="fi-badge flex items-center justify-center gap-x-1 rounded-md text-xs font-medium ring-1 ring-inset px-2 min-w-[theme(spacing.6)] py-1 fi-color-custom bg-custom-50 text-custom-600 ring-custom-600/10 dark:bg-custom-400/10 dark:text-custom-400 dark:ring-custom-400/30 fi-color-primary" style="--c-50:var(--primary-50);--c-400:var(--primary-400);--c-600:var(--primary-600);">
+                                <span class="grid">
+                                <span class="truncate">%s</span>
+                                </span>
+                                </span>
+                                </a>',
```

Comment:
This is repeated a bunch. Is there a way to prevent that? Store this as a string somewhere to be re-used?

Reply:
Yes totally a bad practice on my part. Refactored both bre field and bre rules to remove duplicate code.